site/app/templates/misc/MarkdownArea.twig

```diff
@@ -109,14 +109,54 @@
     </div>
 </div>
 <script>
-    $('#{{markdown_area_id}}').ready(function() {
-        enableTabsInTextArea("#{{markdown_area_id}}");
-        const markdown_area = $('#{{markdown_area_id}}').closest('.markdown-area')
-        markdown_area.find('[data-initialize-preview="1"]').trigger('click');
+$(document).ready(function() {
+    const MIN_HEIGHT = 100;
+
+    function resizeTextarea(textarea) {
+        // Temporarily reduce padding to measure the natural content height more accurately
+        var originalPadding = getComputedStyle(textarea).padding;
+        textarea.style.padding = '0';
+        textarea.style.height = `${MIN_HEIGHT}px`;
+        var desiredHeight = Math.max(textarea.scrollHeight, MIN_HEIGHT);
+        // Restore original padding
+        textarea.style.padding = originalPadding;
+        
+        if (desiredHeight > MIN_HEIGHT) {
+            desiredHeight += 5;
+            textarea.style.height = desiredHeight + 'px';
+        }
+
+        textarea.style.overflowY = 'hidden';
+    }
+    $('textarea').on('input', function() {
+        resizeTextarea(this);
     });
-    $('.markdown-mode-tab').on('click', function(){
-        $(this).addClass('active');
-        const markdown_area = $(this).closest('.markdown-area');
-        markdown_area.find('.markdown-mode-tab').not(this).removeClass('active');
+    $(document).ajaxComplete(function() {
+        $('textarea').each(function() {
+            resizeTextarea(this);
+        });
     });
-</script>
\ No newline at end of file
+
+    // Resize textareas when the content is dynamically added
+    var observer = new MutationObserver(function(mutations) {
+        mutations.forEach(function(mutation) {
+            mutation.addedNodes.forEach(function(node) {
+                if (node.nodeType === 1 && node.matches('textarea')) {
+                    resizeTextarea(node);
+                }
+            });
+        });
+    });
```

Comment:
This will look for any textarea and update that. We probably only want to focus on the markdown area that this is responsible for. Otherwise it could impact other areas which we don't expect. If we desire this behavior everywhere, it should be in a more global spot. If you plan to just keep this here then add an extra check to ensure the element is the correct ID.